capz/readme.md

```diff
@@ -4,44 +4,96 @@ These scripts and templates are used with [cluster-api-provider-azure](https://g
+
+> Note: Look at the `extra_refs` section of the **ci-kubernetes-e2e-capz-master-windows** in [release-master-windows.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml) to see which branches the SIG-Windows e2e test passes are using during the periodic jobs.
+
+### Set environment variables
+
+#### Required
 
 ```bash
 export AZURE_SUBSCRIPTION_ID=<sub-id>
 export AZURE_CLIENT_ID=<client-id>
 export AZURE_CLIENT_SECRET=<client-secret>
 export AZURE_TENANT_ID=<tenantid>
-export CAPZ_DIR="$HOME/<path-to-capz>/cluster-api-provider-azure
+export CAPZ_DIR="$HOME/<path-to-repo>/cluster-api-provider-azure"
+export AZURE_CLOUD_PROVIDER_ROOT="$HOME/<path-to-repo>/cloud-provider-azure"
+
+# run-capz-e2e.sh builds and publishes cloud-provider-azure container images
+# to the registry specified below.  Ensure you are logged in to the registry
+# prior to running the script!
+# TODO: Figure out how to use pre-built images here...
+export REGISTRY="<registry>"
 
 # optional for ability to use your own ssh key (otherwise it generates one)
 # NOTE: Azure does not support ed25519 encrypted SSH keys!
 export AZURE_SSH_PUBLIC_KEY_FILE="$HOME/.ssh/id_rsa.pub"
 ```
 
-## Other configuration
+#### Optional
 
 | ENV variable  | Description  |
 | ------------- | ------------ |
+| `API_SERVER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-apiserver |
+| `AZURE_LOCATION` | The azure region to deploy resources into |
+| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.xx` where x is valid kubernetes minor version such as `latest-1.24` |
+| `NODE_FEATURE_GATES` | Comma-seperated list of feature-gates and their values to pass to the kubelet |
+| `NODE_MACHINE_TYPE` | The [Azure vm size](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes) to use for the nodes |
+| `RUN_SERIAL_TESTS` | If set to `true` then serial slow tests will be run with default ginkgo settings |
 | `SKIP_CREATE` | Don't create a cluster.  Must set `CLUSTER_NAME` and have current a workload cluster kubeconfig file with name `./"${CLUSTER_NAME}".kubeconfig` |
 | `SKIP_LOG_COLLECTION` | Don't collect logs from the cluster |
 | `SKIP_TEST`  | Only creates the cluster, will not run tests |
 | `SKIP_CLEANUP` | Don't delete the cluster / resource group after script executions |
-| `RUN_SERIAL_TESTS` | If set to `true` then serial slow tests will be run with default ginkgo settings |
-| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.x` where x is valid kubernetes minor version such as `latest-1.24` |
-| `AZURE_LOCATION` | The azure region to deploy resources into |
+| `SCHEDULER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-scheduler |
+| `WINDOWS_CONTAINERD_URL` | URL to a containerd release tarball to use for the Windows nodes |
```

Comment:
What happens if this is not set?

Reply:
it is defaulted https://github.com/kubernetes-sigs/windows-testing/blob/058c4d333aa213f488dbf141a029beeda8dd2f27/capz/run-capz-e2e.sh#L27